### DISCUSSION TITLE
Hydrogen/ammonia mechanism suitable for PeleLMeX

### DISCUSSION THREAD

**Discussion by Alex-jian522** [AMReX-Combustion/PeleLMeX#349] [General]
Hello,
I followed the tutorial (https://amrex-combustion.github.io/PelePhysics/Ceptr.html#converting-chemkin-files) to convert the hydrogen/ammonia mechanism to one suitable for use with PeleLMeX. But when testing one-dimensional flames, they often diverge due to chemical stifness.
I would like to ask if there is any recommendation for hydrogen/ammonia mechanism suitable for PeleLMeX?
Thank you so much!

**Comment by Alex-jian522** [comment]
Now I guess this problem is not resulted from the mechanism. As I have change three different mechanisms and the problem is the same.
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/72745362/b13022da-829d-404d-ae43-49f996910aff)
The case is a 1D freely propagating NH3/H2/air flame, but I set it as 2D. 
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/72745362/1b86056d-0971-4ac4-a060-044e326f8c83)

I have tried decreased CFL number to 0.05 or use fixed time step (1e-7)， but it never works.

**Comment by esclapez** [comment]
The spatial resolution can be as critical as the temporal one, especially upon initialization. Did you try reducing the domain size or adding AMR levels to have a better resolution of the flame ? Which CVODE option are you using, so are more robust (albeit possibly more expensive) than others ?

**Reply by Alex-jian522** [reply]
Yes, I have try to add AMR levels, but the same error still happen. The CVODE is set as bellow:
peleLM.chem_integrator = "ReactorCvode"
peleLM.use_typ_vals_chem = 1          # Use species/temp typical values in CVODE
ode.rtol = 1.0e-6                     # Relative tolerance of the chemical solve
ode.atol = 1.0e-5                     # Absolute tolerance factor applied on typical values
cvode.solve_type = GMRES               # CVODE Linear solve type (for Newton direction)
cvode.max_order  = 4                   # CVODE max BDF order.

**Reply by esclapez** [reply]
Can you try:
```
cvode.solve_type = denseAJ_direct
```
the analytical Jacobian (AJ) should have been generated by CEPTR when you imported this chemical mechanism into PelePhysics, and it might be more robust (and generally faster for stiff mechanisms too).
